pymc_experimental/model_builder.py

```diff
@@ -187,13 +187,10 @@ def default_sampler_config(self) -> Dict:
         raise NotImplementedError
 
     @abstractmethod
-    def generate_and_preprocess_model_data(
-        self, X: Union[pd.DataFrame, pd.Series], y: pd.Series
-    ) -> None:
+    def preprocess_model_data(self, X: Union[pd.DataFrame, pd.Series], y: pd.Series = None) -> None:
```

Comment:
Signature doesn't match the return value from docstring, will result in mypy error in child class about breaking Liskov's principle